### PR TITLE
Connections: Pass keywords from the backend to CommandPalette

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- adds `keywords` to the backend navModel
- passes these through as `keywords` to the command palette actions

**Why do we need this feature?**

- better ux to find how to create new connections

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

For https://github.com/grafana/grafana/issues/78158

**Special notes for your reviewer:**

- this allows searching for e.g. `prometheus` in the command palette and matching the 'Add new connection' page. 
- this **does not** prepopulate the 'Add new connection' search. that will come in a later PR. 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/navtree/navtreeimpl/navtree.go [hunk 1]
```diff
@@ -500,6 +500,7 @@ func (s *ServiceImpl) buildDataConnectionsNavLink(c *contextmodel.ReqContext) *n
 			SubTitle: "Browse and create new connections",
 			Url:      baseUrl + "/add-new-connection",
 			Children: []*navtree.NavLink{},
+			Keywords: []string{"graphite", "loki", "prometheus", "sql", "tempo"},
```
Comment (ashharrison90) [inline pkg/services/navtree/navtreeimpl/navtree.go:503]: @mikkancso not sure exactly which datasources you want to add here? i added the basics, but we can always add more 🤔

Comment (mikkancso) [reply pkg/services/navtree/navtreeimpl/navtree.go:503]: Based on [this dashboard](https://bi.grafana-ops.net/d/I020tGa4k/data-source-search-term-frequencies?orgId=1&from=now-30d&to=now), I'd also add "json" and "csv".

**Review by mikkancso — COMMENTED** [review]
Thanks! :pray:

**Review by eledobleefe — APPROVED** [review]
Looks 👌 to me

**Review by joshhunt — APPROVED** [review]
Works great!

**Pull request merged**